owncloudApp/src/main/java/com/owncloud/android/providers/FileContentProvider.kt

```diff
@@ -0,0 +1,1470 @@
+            mShareProjectionMap[ProviderTableMeta.OCSHARES_EXPIRATION_DATE] = ProviderTableMeta.OCSHARES_EXPIRATION_DATE
+            mShareProjectionMap[ProviderTableMeta.OCSHARES_TOKEN] = ProviderTableMeta.OCSHARES_TOKEN
+            mShareProjectionMap[ProviderTableMeta.OCSHARES_SHARE_WITH_DISPLAY_NAME] =
+                    ProviderTableMeta.OCSHARES_SHARE_WITH_DISPLAY_NAME
+            mShareProjectionMap[ProviderTableMeta.OCSHARES_SHARE_WITH_ADDITIONAL_INFO] =
+                    ProviderTableMeta.OCSHARES_SHARE_WITH_ADDITIONAL_INFO
+            mShareProjectionMap[ProviderTableMeta.OCSHARES_IS_DIRECTORY] = ProviderTableMeta.OCSHARES_IS_DIRECTORY
+            mShareProjectionMap[ProviderTableMeta.OCSHARES_USER_ID] = ProviderTableMeta.OCSHARES_USER_ID
+            mShareProjectionMap[ProviderTableMeta.OCSHARES_ID_REMOTE_SHARED] =
+                    ProviderTableMeta.OCSHARES_ID_REMOTE_SHARED
+            mShareProjectionMap[ProviderTableMeta.OCSHARES_ACCOUNT_OWNER] = ProviderTableMeta.OCSHARES_ACCOUNT_OWNER
+            mShareProjectionMap[ProviderTableMeta.OCSHARES_NAME] = ProviderTableMeta.OCSHARES_NAME
+            mShareProjectionMap[ProviderTableMeta.OCSHARES_URL] = ProviderTableMeta.OCSHARES_URL
+        }
+
+        private val mCapabilityProjectionMap = HashMap<String, String>()
+
+        init {
+
+            mCapabilityProjectionMap[ProviderTableMeta._ID] = ProviderTableMeta._ID
+            mCapabilityProjectionMap[ProviderTableMeta.CAPABILITIES_ACCOUNT_NAME] =
+                    ProviderTableMeta.CAPABILITIES_ACCOUNT_NAME
+            mCapabilityProjectionMap[ProviderTableMeta.CAPABILITIES_VERSION_MAYOR] =
+                    ProviderTableMeta.CAPABILITIES_VERSION_MAYOR
+            mCapabilityProjectionMap[ProviderTableMeta.CAPABILITIES_VERSION_MINOR] =
+                    ProviderTableMeta.CAPABILITIES_VERSION_MINOR
+            mCapabilityProjectionMap[ProviderTableMeta.CAPABILITIES_VERSION_MICRO] =
+                    ProviderTableMeta.CAPABILITIES_VERSION_MICRO
+            mCapabilityProjectionMap[ProviderTableMeta.CAPABILITIES_VERSION_STRING] =
+                    ProviderTableMeta.CAPABILITIES_VERSION_STRING
+            mCapabilityProjectionMap[ProviderTableMeta.CAPABILITIES_VERSION_EDITION] =
+                    ProviderTableMeta.CAPABILITIES_VERSION_EDITION
+            mCapabilityProjectionMap[ProviderTableMeta.CAPABILITIES_CORE_POLLINTERVAL] =
+                    ProviderTableMeta.CAPABILITIES_CORE_POLLINTERVAL
+            mCapabilityProjectionMap[ProviderTableMeta.CAPABILITIES_SHARING_API_ENABLED] =
+                    ProviderTableMeta.CAPABILITIES_SHARING_API_ENABLED
+            mCapabilityProjectionMap[ProviderTableMeta.CAPABILITIES_SHARING_PUBLIC_ENABLED] =
+                    ProviderTableMeta.CAPABILITIES_SHARING_PUBLIC_ENABLED
+            mCapabilityProjectionMap[ProviderTableMeta.CAPABILITIES_SHARING_PUBLIC_PASSWORD_ENFORCED
```

Comment:
what a party this file!!! I know it's tech debt, but please, from now on, "divide and rule"

Reply:
Yes, I agree with you, hope that part of this file is reduced when migration to room is completed